admin/lib/generators/solidus_admin/component/component_generator.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SolidusAdmin
+  class ComponentGenerator < Rails::Generators::NamedBase
+    source_root File.expand_path('templates', __dir__)
+
+    argument :attributes, type: :array, default: [], banner: "attribute"
+
+    class_option :html, type: :boolean, default: true
+    class_option :i18n, type: :boolean, default: true
+    class_option :js, type: :boolean, default: true
+    class_option :spec, type: :boolean, default: true
```

Comment:
If I recall correctly, we are still undecided if we want to keep the component's tests in the sidecar directory. Do we want to start the discussion here in this context? Alternatively, we can extract this part in another PR/issue to facilitate merging this one.